packages/ssh-tunnel/src/index.ts

```diff
@@ -195,7 +144,53 @@ export class SshTunnel extends EventEmitter {
     }
   }
 
+  private async connectSsh(): Promise<void> {
+    if (this.connected) {
+      debug('already connected');
+      return;
+    }
+
+    if (this.connectingPromise) {
+      debug('reusing connectingPromise');
+      return this.connectingPromise;
+    }
+
+    if (this.closed) {
+      // A socks5 request could come in after we deliberately closed the connection. Don't reconnect in that case.
+      throw new Error('Disconnected.');
+    }
+
+    debug('creating SSH connection');
+    const ac = new AbortController();
+
+    this.connectingPromise = Promise.race([
+      once(this.sshClient, 'error', { signal: ac.signal }).then(([err]) => {
+        throw err;
+      }),
+      (() => {
+        const waitForReady = once(this.sshClient, 'ready', { signal: ac.signal }).then(() => { return; });
```

Comment:
```suggestion
        const waitForReady = once(this.sshClient, 'ready', { signal: ac.signal });
```

Reply:
waitForReady isn't `Promise<void>`. There was some `Promise<[void]>` going on that I'm trying to figure out how to cast.

Reply:
I think in that case I’d just do

```suggestion
        const waitForReady = once(this.sshClient, 'ready', { signal: ac.signal }).then(() => {});
```

since otherwise the `return;` looks a bit out-of-place, as if the intention was to actually return something from the `.then()` callback

Reply:
Ugh. Yeah, this is fine, but still icky that we need to do this (what’s wrong with empty arrow functions?) :confused: